doc/explanation/mental-model.rst

```diff
@@ -0,0 +1,175 @@
+The Dune Mental Model
+=====================
+
+It is not strictly necessary to understand Dune's underlying model to use it;
+but knowing how it works under the hood will help understand some errors and
+what's possible with Dune.
+
+.. note::
+
+   This document is a simplification of the reality: the actual rules might be
+   different, it does not touch rule loading and glosses over how caching
+   works, but should be a useful tool to build an understanding of Dune.
+
+How Dune Works
+--------------
+
+The building brick of Dune is the *rule*:
+
+    A *rule* reads *dependencies* and writes *targets* by executing an *action*.
+
+When ``dune build`` is executed, it will first read ``dune`` files to determine
```

Comment:
```suggestion
When ``dune build`` is executed, it will first read the project's ``dune`` files to determine
```

As it feels a bit like there's a word missing.